### PR TITLE
Update installer for October L6 upgrade.

### LINKED ISSUE #1: Update for Laravel 6
@bennothommo Could you review this installer and make sure it's setup for the L6 upgrade in regards to system requirements? Might also want to update the embedded CA files / investigate whether those are actually necessary. It would be nice if it supported checking for all the requirements the web installer did.

### PR DESCRIPTION
**Pull request by bennothommo**
- Updated dependency checks for PHP and extensions.
- Updated bundled CA certificates
- Updated TLS cipher list
- Prevented installer from affecting Composer files
- Remove installer after completing download of October CMS

Fixes #1.

### CODE REVIEW

File: installer.php [hunk 1]
```diff
@@ -212,86 +210,75 @@ function getPlatformIssues(&$errors, &$warnings)
...
 
     if (!ini_get('allow_url_fopen')) {
         $errors['allow_url_fopen'] = array(
-            'The allow_url_fopen setting is incorrect.',
+            'The `allow_url_fopen` setting is incorrect.',
             'Add the following to the end of your `php.ini`:',
             '    allow_url_fopen = On',
             $iniMessage
         );
     }
```
Comment (LukeTowers) [inline installer.php:255]: Should we add this check to the web installer?

File: installer.php [hunk 2]
```diff
@@ -212,86 +210,75 @@ function getPlatformIssues(&$errors, &$warnings)
...
 
     if (!extension_loaded('iconv') && !extension_loaded('mbstring')) {
         $errors['iconv_mbstring'] = array(
-            'The iconv OR mbstring extension is required and both are missing.',
-            'Install either of them or recompile php without --disable-iconv'
+            'The `iconv` OR `mbstring` extension is required and both are missing.',
+            'Install either of them, or recompile PHP without `--disable-iconv`'
         );
     }
```
Comment (LukeTowers) [inline installer.php:246]: Is this even possible in modern versions of PHP? If it is, should we add it to https://github.com/octobercms/docs/blob/master/setup-installation.md#minimum-system-requirements / the web installer?

File: installer.php [hunk 3]
```diff
@@ -212,86 +210,75 @@ function getPlatformIssues(&$errors, &$warnings)
...
 
     if (!extension_loaded('hash')) {
         $errors['hash'] = array(
-            'The hash extension is missing.',
-            'Install it or recompile php without --disable-hash'
+            'The `hash` extension is missing.',
+            'Install it, or recompile PHP without `--disable-hash`'
         );
     }
```
Comment (LukeTowers) [inline installer.php:239]: Is this even possible in modern versions of PHP? If it is, should we add it to https://github.com/octobercms/docs/blob/master/setup-installation.md#minimum-system-requirements / the web installer?

File: installer.php [hunk 4]
```diff
@@ -212,86 +210,75 @@ function getPlatformIssues(&$errors, &$warnings)
...
 
     if (!extension_loaded('filter')) {
         $errors['filter'] = array(
-            'The filter extension is missing.',
-            'Install it or recompile php without --disable-filter'
+            'The `filter` extension is missing.',
+            'Install it, or recompile PHP without `--disable-filter`'
         );
     }
```
Comment (LukeTowers) [inline installer.php:232]: Is this even possible in modern versions of PHP? If it is, should we add it to https://github.com/octobercms/docs/blob/master/setup-installation.md#minimum-system-requirements / the web installer?

File: installer.php [hunk 5]
```diff
@@ -212,86 +210,75 @@ function getPlatformIssues(&$errors, &$warnings)
...
-    }
-
     if (!function_exists('json_decode')) {
         $errors['json'] = array(
-            'The json extension is missing.',
-            'Install it or recompile php without --disable-json'
+            'The `json` extension is missing.',
+            'Install it, or recompile PHP without `--disable-json`'
         );
     }
```
Comment (LukeTowers) [inline installer.php:218]: Is this even possible in modern versions of PHP?

File: installer.php [hunk 6]
```diff
@@ -212,86 +210,75 @@ function getPlatformIssues(&$errors, &$warnings)
...
     $iniMessage .= PHP_EOL.'If you can not modify the ini file, you can also run `php -d option=value` to modify ini values on the fly. You can use -d multiple times.';
 
-    if (ini_get('detect_unicode')) {
-        $errors['unicode'] = array(
-            'The detect_unicode setting must be disabled.',
-            'Add the following to the end of your `php.ini`:',
-            '    detect_unicode = Off',
-            $iniMessage
-        );
-    }
```
Comment (LukeTowers) [inline installer.php:222]: Can we add checks for all of the extensions listed in https://github.com/octobercms/docs/blob/master/setup-installation.md#minimum-system-requirements?

Comment (LukeTowers) [reply installer.php:222]: Additionally, can we provide a recommended command for installing the required missing extension to the warning message?

File: installer.php [hunk 7]
```diff
@@ -25,7 +25,6 @@ function process($argv)
     }
 
     $check      = in_array('--check', $argv);
-    $help       = in_array('--help', $argv);
```
[the remark this answers is not in the document]
Comment (bennothommo) [reply installer.php:28]: @LukeTowers it's a dead variable - the `--help` argument is checked above it.

File: installer.php (hunk 1 above)
Comment (bennothommo) [reply installer.php:255]: Good idea - I'll submit a PR to that too.

File: installer.php (hunk 2 above)
Comment (bennothommo) [reply installer.php:246]: I found out with Manjaro Linux that it is possible - some Linux distributions don't enable *any* extensions except for those necessary for PHP itself. `iconv` is technically one of them, but even that can be disabled, so the check makes sense even if 99.9999% of people won't encounter it.

File: installer.php (hunk 3 above)
Comment (bennothommo) [reply installer.php:239]: Same as above.

File: installer.php (hunk 4 above)
Comment (bennothommo) [reply installer.php:232]: Same as above.

File: installer.php (hunk 5 above)
Comment (bennothommo) [reply installer.php:218]: Same as above.

File: installer.php (hunk 6 above)
Comment (bennothommo) [reply installer.php:222]: Yep, sounds good.

File: installer.php (hunk 2 above)
Comment (bennothommo) [reply installer.php:246]: The `mbstring` extension is already in the minimum requirements, and the web installer checks it here: https://github.com/octobercms/install/blob/master/install_files/php/Installer.php#L85

File: installer.php (hunk 1 above)
Comment (bennothommo) [reply installer.php:255]: https://github.com/octobercms/install/pull/107

File: installer.php (hunk 3 above)
Comment (bennothommo) [reply installer.php:239]: https://github.com/octobercms/install/pull/107

File: installer.php (hunk 4 above)
Comment (bennothommo) [reply installer.php:232]: https://github.com/octobercms/install/pull/107

**Pull request merged**